src/modules/terminalfont.c

```diff
@@ -69,6 +67,51 @@ static void printXFCE4Terminal(FFinstance* instance)
     printTerminalFont(instance, font);
 }
 
+static void printTilixTerminal(FFinstance* instance)
+{
+    static const char* fontName = NULL;
+    static const char* defaultProfile = NULL;
+    static FFstrbuf key;
+    static FFvariant res;
+    res.strValue = NULL;
+    ffStrbufInitAS(&key, 64, "/com/gexperts/Tilix/profiles/");
+
+    defaultProfile = (ffSettingsGetGsettings(instance, "com.gexperts.Tilix.ProfilesList", NULL, "default", FF_VARIANT_TYPE_STRING)).strValue;
+
+    if (!defaultProfile)
+    {
+        ffPrintError(instance, FF_TERMFONT_MODULE_NAME, 0, &instance->config.termFontKey, &instance->config.termFontFormat, FF_TERMFONT_NUM_FORMAT_ARGS, "Couldn't find \"Default\" profile in Tilix settings");
+        ffStrbufDestroy(&key);
+        return;
+    }
+    else
+    {
+        ffStrbufAppendS(&key, defaultProfile);
+        ffStrbufAppendS(&key, "/use-system-font");
+
+        res = ffSettingsGetDConf(instance,key.chars, FF_VARIANT_TYPE_BOOL);
+
+        if(res.boolValueSet && res.boolValue == 0) // custom font
+        {
+            ffStrbufRemoveStrings(&key, 1, "/use-system-font");
```

Comment:
Removing strings is very slow. Maybe store the length of key before `ffStrbufAppendS(&key, "/use-system-font")` and use a `ffSubstrBefore` here.